sechub-sereco/src/main/java/com/daimler/sechub/sereco/importer/SarifV1JSONImporter.java

```diff
@@ -51,93 +59,196 @@ public SerecoMetaData importResult(String data) throws IOException {
-        int index = name.lastIndexOf("/");
-        if (index != -1) {
-            index++;
-            if (index < name.length() - 1) {
-                name = name.substring(index);
+            if (CWE.equalsIgnoreCase(toolComponentName)) {
+                /* cwe found, so lets look after the id */
+                data.cweId = id;
             }
         }
-        return name;
     }
 
-    private SarifSchema210 readSarifSchema210(String data) throws JsonProcessingException, JsonMappingException {
-        return mapper.readValue(data, SarifSchema210.class);
+    private String resolveDescription(Rule rule) {
+        return rule.getFullDescription().getText();
+    }
+
+    private String resolveType(Rule rule, Run run) {
+        if (rule == null) {
+            return "error:rule==null!";
+        }
+        String type = null;
+        Message shortDescription = rule.getShortDescription();
+        if (shortDescription != null) {
+            type = shortDescription.getText();
+        }
+
+        if (type == null) {
+            type = handleSpecialToolTypeResolving(rule, run);
+        }
+
+        if (type == null) {
+            /*
+             * no type identifier found, so do fallback to id, we do not use "name" becaus
+             * this is for i18n!
+             */
+            type = rule.getId();
+        }
+        return type;
+    }
+
+    private String handleSpecialToolTypeResolving(Rule rule, Run run) {
+        Tool tool = run.getTool();
+        if (tool == null) {
+            return null;
+        }
+        Driver driver = tool.getDriver();
+        if (driver == null) {
+            return null;
+        }
+        String toolName = driver.getName();
+        if (toolName == null) {
+            return null;
+        }
+        if (toolName.equalsIgnoreCase("Brakeman")) {
+            return handleSpecialBrakemanResolving(rule, run);
+        }
+        return null;
+    }
+
+    private String handleSpecialBrakemanResolving(Rule rule, Run run) {
+        /*
+         * At this point, we MUST have tried before to fetch the type by short
+         * description from SARIF JSON report file, which is normally set by many tools.
+         * 
+         * If this is not possible we normally do a fallback to the identifier.
+         * 
+         * In Brakeman 5.0.0 IDs starting with "BRAKE" - e.g. BRAKE0102, but we cannot
+         * use this id for our type information in Serec, because it's an internal ID.
+         * So not really human readable and you even cannot "just google it".
+         * 
+         * But it does provide a name represented by "ContentTag/$shortDescription" -
+         * for example for ID "BRAKE0102" the name is "ContentTag/Cross-Site Scripting".
+         * 
+         * Inside SARIF specification the name field is meant to be for i18n - but we
+         * handle this special for brakeman here as long as there is no shortDescription
```

Comment:
We can create a ticket and inform Brakeman about this mistake: https://github.com/presidentbeef/brakeman. However, there is really no reason to give a tool a special treatment, just because there implementation has some issues.

Especially, because one can use a Brakeman converter in the PDS to fix issues like this and put the report into a proper format.

Reply:
I removed the brakeman special treatment - especially because of the mentioned license issue.